server/src/main/java/org/apache/iotdb/db/mpp/execution/executor/RegionWriteExecutor.java

```diff
@@ -0,0 +1,470 @@
+      }
+      return response;
+    }
+
+    private ConsensusWriteResponse executePlanNodeInConsensusLayer(
+        ConsensusGroupId groupId, PlanNode planNode) {
+      if (groupId instanceof DataRegionId) {
+        return DataRegionConsensusImpl.getInstance().write(groupId, planNode);
+      } else {
+        return SchemaRegionConsensusImpl.getInstance().write(groupId, planNode);
+      }
+    }
+
+    @Override
+    public RegionExecutionResult visitInsertRow(
+        InsertRowNode node, WritePlanNodeExecutionContext context) {
+      return executeDataInsert(node, context);
+    }
+
+    @Override
+    public RegionExecutionResult visitInsertTablet(
+        InsertTabletNode node, WritePlanNodeExecutionContext context) {
+      return executeDataInsert(node, context);
+    }
+
+    @Override
+    public RegionExecutionResult visitInsertRows(
+        InsertRowsNode node, WritePlanNodeExecutionContext context) {
+      return executeDataInsert(node, context);
+    }
+
+    @Override
+    public RegionExecutionResult visitInsertMultiTablets(
+        InsertMultiTabletsNode node, WritePlanNodeExecutionContext context) {
+      return executeDataInsert(node, context);
+    }
+
+    @Override
+    public RegionExecutionResult visitInsertRowsOfOneDevice(
+        InsertRowsOfOneDeviceNode node, WritePlanNodeExecutionContext context) {
+      return executeDataInsert(node, context);
+    }
+
+    private RegionExecutionResult executeDataInsert(
+        InsertNode insertNode, WritePlanNodeExecutionContext context) {
+      RegionExecutionResult response = new RegionExecutionResult();
+      // data insertion should be blocked by data deletion, especially when deleting timeseries
+      context.getRegionWriteValidationRWLock().readLock().lock();
+      try {
+        try {
+          SchemaValidator.validate(insertNode);
+        } catch (SemanticException e) {
+          response.setAccepted(false);
+          response.setMessage(e.getMessage());
+          if (e.getCause() instanceof IoTDBException) {
+            IoTDBException ioTDBException = (IoTDBException) e.getCause();
+            response.setStatus(
+                RpcUtils.getStatus(ioTDBException.getErrorCode(), ioTDBException.getMessage()));
+          } else {
+            response.setStatus(RpcUtils.getStatus(TSStatusCode.METADATA_ERROR, e.getMessage()));
+          }
+          return response;
+        }
+        boolean hasFailedMeasurement = insertNode.hasFailedMeasurements();
+        String partialInsertMessage = null;
+        if (hasFailedMeasurement) {
+          partialInsertMessage =
+              String.format(
+                  "Fail to insert measurements %s caused by %s",
+                  insertNode.getFailedMeasurements(), insertNode.getFailedMessages());
+          LOGGER.warn(partialInsertMessage);
+        }
+
+        ConsensusWriteResponse writeResponse =
+            executePlanNodeInConsensusLayer(context.getRegionId(), insertNode);
```

Comment:
We can use `DataRegionConsensusImpl.getInstance().write(groupId, planNode)` to avoid ambiguity because current method is only used to process the PlanNode for DataRegion

Reply:
Fixed.